inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -581,6 +581,16 @@ QueryNetworkResult clDNNEngine::QueryNetwork(const CNNNetwork& network,
                     }
                 }
             }
+
+            if (ngraph::op::is_constant(node) || ngraph::op::is_parameter(node)) {
+                if (!InferenceEngine::details::contains(supported, node->output(0).get_target_inputs().begin()->get_node()->get_friendly_name())) {
+                    supported.erase(node->get_friendly_name());
+                }
+            } else if (ngraph::op::is_output(node)) {
+                if (!InferenceEngine::details::contains(supported, node->input_values().begin()->get_node()->get_friendly_name())) {
+                    supported.erase(node->get_friendly_name());
+                }
+            }
```

Comment:
@vladimir-paramuzov I believe this logic fixes the issue when GPU plugin returns only params , results from QueryNetwork. So, you don't need WA which says that Parameter is not supported, now it's optimized by this code